java/arcs/core/storage/StorageProxy.kt

```diff
@@ -575,6 +579,12 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
         }
     }
 
+    private fun checkInDispatcher() = check(
+        !ArcsStrictMode.strictHandles || scheduler.currentlyRunningInSchedulerDispatcher(dispatcher)
```

Comment:
The scheduler knows its own dispatcher, curious why we need to pass it here?

Something like
```kotlin
!ArcsStrictMode.strictHandles || scheduler.isCurrentDispatcher()
```

could suffice

Reply:
done